### PR TITLE
Add missing unit test for resource resize policy defaulting

### PR DESCRIPTION
**Pull request by vinaykul**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change? no

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/assign @thockin

### CODE REVIEW

File: pkg/apis/core/v1/defaults_test.go [hunk 1]
```diff
@@ -1906,3 +1907,162 @@ func TestSetDefaultServiceInternalTrafficPolicy(t *testing.T) {
...
+					ResourceName:  v1.ResourceCPU,
+					RestartPolicy: v1.RestartContainer,
+				},
+				{
+					ResourceName:  v1.ResourceMemory,
+					RestartPolicy: v1.NotRequired,
+				},
+			},
+		},
+		"CPU and memory requests are specified with restartContainer resize policy for both": {
```
Comment (SergeyKanzhelev) [inline pkg/apis/core/v1/defaults_test.go:2027]: can you add a test for some other resource name other than memory and CPU

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

**Comment by vinaykul** [comment]
/assign @sftim 
Please review changelog, thanks.

**Comment by SergeyKanzhelev** [comment]
Were api/openapi-spec/swagger.json and api/openapi-spec/v3/apis__storage.k8s.io__v1beta1_openapi.json picked up by mistake?

Changelog and unit tests lgtm

**Comment by vinaykul** [comment]
> Were api/openapi-spec/swagger.json and api/openapi-spec/v3/apis__storage.k8s.io__v1beta1_openapi.json picked up by mistake?
> 
> Changelog and unit tests lgtm

k8s-verify job failed and failure logs asked to run update-openapi-spec. https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/116684/pull-kubernetes-verify/1636573699786149888

It is unrelated to my changes but without it CI blocks the PR.

File: api/openapi-spec/swagger.json [hunk 2]
```diff
@@ -14460,89 +14460,6 @@
       },
       "type": "object"
     },
```
Comment (pacoxu) [inline api/openapi-spec/swagger.json:14462]: This is fixed in https://github.com/kubernetes/kubernetes/pull/116705 and you may wait for that change to be merged and rebase with the latest code later.

Comment (vinaykul) [reply api/openapi-spec/swagger.json]: rebased and updated, thanks.

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**